packages/core/src/Primitive.tsx

```diff
@@ -0,0 +1,36 @@
+import React from 'react';
+import type { CommonProps } from './types';
+
+type PrimitiveOwnProps<E extends React.ElementType = React.ElementType> = {
+  as?: E;
+} & CommonProps;
+
+export type PrimitiveProps<E extends React.ElementType> = PrimitiveOwnProps<E> &
+  Omit<React.ComponentProps<E>, keyof PrimitiveOwnProps>;
```

Comment:
Can we use `Pick` here? It's more useful to see the list of picked properties in VS Code than the ones excluded.

Reply:
I don't think we can use Pick here cause we don't really know what to pick :) It's a generic type and we can only say what shouldn't be included in the type.

Reply:
Something like `Pick<React.ComponentProps<E>, Exclude<React.ComponentProps<E>, keyof PrimitiveOwnProps>>;` ?